src/common/api/accessibilityClient.js

```diff
@@ -0,0 +1,91 @@
+import { ApiClient } from './client';
+import constants from '../../../app/constants/AppConstants';
+
+export class AccessibilityApiClient {
+  apiClient;
+
+  systemId;
+
+  constructor(baseUrl, systemId) {
+    this.apiClient = new ApiClient(baseUrl, false);
+    this.systemId = systemId;
+  }
+
+  getAccessibilityInformation(resourceId, locale = 'fi') {
+    return Promise.all([
+      this.getSentences(resourceId, locale),
+      this.getShortcomings(resourceId, locale),
+    // eslint-disable-next-line
+    ]).then(([details, shortcomings]) => (details.length > 0 || shortcomings.length > 0 ? { details, shortcomings } : undefined));
+  }
+
+  getSentences(resourceId, locale = 'fi') {
+    return this.apiClient
+      .get(`targets/${this.systemId}/${resourceId}/sentences`)
+      .then((res) => {
+        return res.data
+          .map((sentenceObject) => {
+            return {
+              sentenceGroup: sentenceObject.sentenceGroups.find(
+                sentenceGroup => sentenceGroup.language === locale,
+              ).value,
+              sentences: sentenceObject.sentences
+                .filter(sentence => sentence.language === locale)
+                .map(sentence => sentence.value),
+            };
+          })
+          .reduce((acc, sentenceObject) => {
+            const index = acc.findIndex(
+              accSentence => accSentence.sentenceGroup === sentenceObject.sentenceGroup,
+            );
+
+            if (index >= 0) {
+              acc[index] = {
+                ...acc[index],
+                sentences: [
+                  ...acc[index].sentences,
+                  ...sentenceObject.sentences,
+                ],
+              };
+              return acc;
+            }
+
+            return [...acc, sentenceObject];
+          }, []);
+      });
+  }
+
+  getShortcomings(resourceId, locale = 'fi') {
+    // eslint-disable-next-line
+    return Promise.all([this.apiClient.get(`targets/${this.systemId}/${resourceId}/shortages`), this.apiClient.get('viewpoints')])
+      .then(([shortagesRes, viewpointsRes]) => {
+        const idToNameMap = {};
+
+        viewpointsRes.data.forEach((viewpoint) => {
+          idToNameMap[viewpoint.viewpointId] = viewpoint.names.find(name => name.language === locale).value;
```

Comment:
In some edge cases, `find` return value might be `undefined`.

Reply:
Added a function to resolve fallback locales (defaults to fi).